### PR TITLE
Update Gradle, Loom for 1.8.9

### PR DESCRIPTION
**Pull request by halotroop2288**
A bunch of changes to the build.gradle.
- Updated Gradle to 7.0
- Updated Loom to `0.7-SNAPSHOT` (Latest version that supports Java 8)
- Moved global variable declaration back to gradle.properties.
- Solved a few warnings by using closures
- Fixed a bug that caused building to fail during `processResources` task

### CODE REVIEW

File: src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -5,18 +5,34 @@
...
     "sources": "https://github.com/Legacy-Fabric/fabric"
   },
   "authors": [
-    "Legacy Fabric",
-    "FabricMC"
+    {
+      "name": "Legacy Fabric/Quilt",
+      "contact": {
+        "homepage": "https://legacyfabric.net/",
+        "discord": "https://discord.com/invite/yBTpRXe",
```
Comment (Grayray75) [inline src/main/resources/fabric.mod.json:19]:
```suggestion
        "discord": "https://legacyfabric.net/discord",
```
Link the permanent invite

Comment (halotroop2288) [reply src/main/resources/fabric.mod.json]: Done.

File: build.gradle [hunk 2]
```diff
@@ -3,26 +3,22 @@ plugins {
...
+def ENV = System.getenv()
 
-version = Globals.baseVersion + "+" + Globals.mcVersion
+setVersion("${project.base_version}+${project.mc_version}")
 logger.lifecycle("Building Fabric: " + version)
 
-import org.apache.commons.codec.digest.DigestUtils
+
+import net.fabricmc.loom.task.RemapJarTask
+import org.apache.commons.codec.digest.
```
Comment (Grayray75) [inline build.gradle:20]: This doesn't seem right

Comment (halotroop2288) [reply build.gradle:20]: Well it is. There was an unnecessary qualified reference.

**Comment by halotroop2288** [comment]
Approving that workflow would be nice right now, so I can check that I haven't messed anything up by making edits via Github.

**Review by arthurbambou — APPROVED** [review]

**Review by TehcJS — APPROVED** [review]

**Review by TehcJS — COMMENTED** [review]
icon got crabbed

**Review by TehcJS — APPROVED** [review]

**Comment by BluCobalt** [comment]
Kinda weary about Gradle 7, I'll check the workflows though

**Comment by halotroop2288** [comment]
> Kinda weary about Gradle 7 because it's supposed to break a lot of things, but I'll check the workflows

You're right to be weary, but fear not. I have fixed these issues.....

But Git broke them again. It removed random stuff from the file, and I only find this out after it runs the workflow.... Can we run it again to check that I've finally fixed it.

File: build.gradle [hunk 3]
```diff
@@ -3,27 +3,23 @@ plugins {
 	id 'eclipse'
 	id 'idea'
 	id 'maven-publish'
-	id 'fabric-loom' version '0.6-SNAPSHOT'
-	id 'net.minecrell.licenser' version '0.4.1'
+	id 'fabric-loom' version "0.7-SNAPSHOT"
+	id 'net.minecrell.licenser' version "0.4.1"
```
Comment (BoogieMonster1O1) [inline build.gradle:7]: licenser seems to break checks, maybe this needs updating as well?
the latest is `0.6.0`

**Review by hYdos — APPROVED** [review]

**Comment by BluCobalt** [comment]
Ran it again

File: build.gradle (hunk 3 above)
Comment (halotroop2288) [reply build.gradle]: It didn't work. It says it wasn't found. :(

Comment (BoogieMonster1O1) [reply build.gradle]: looks like the id got updated as well
its `org.cadixdev.licenser` now

**Comment by BluCobalt** [comment]
Ran it again

**Comment by halotroop2288** [comment]
New problem, needs discussion: It won't compile on 1.8 anymore because of a dependency we're not in control of being compiled on a newer version. Should we find a way to downgrade that dependency, or just remove JDK 8 compat (would still work in prod with JRE, probably.)

**Comment by halotroop2288** [comment]
🦀 JDK 1.8. Who needs it anyway?

**Comment by halotroop2288** [comment]
Haha PR with three approvals not being merged go brrrrrrrrrr

**Comment by BluCobalt** [comment]
Is this broken with jdk8?

**Comment by halotroop2288** [comment]
> Is this broken with jdk8?

No. The PR is just fine. Something else beyond our control is broken. It broke pretty much every project I've looked at recently in Legacy Fabric. Spooky.

Luckily, JDK 11 produces compatible bytecode.... According to @hYdos anyway.

**Review by BoogieMonster1O1 — APPROVED** [review]

**Comment by BluCobalt** [comment]
Didn't even let me approve smh

**Comment by hYdos** [comment]
same smh

**Comment by BoogieMonster1O1** [comment]
> Didn't even let me approve smh

image

**Comment by BoogieMonster1O1** [comment]
> same smh

image

**Comment by BluCobalt** [comment]
Bad tm

**Comment by BoogieMonster1O1** [comment]
> Bad tm

image

**Comment by hYdos** [comment]
Bad TMh5

**Comment by halotroop2288** [comment]
Stop necroing. Lock thread pls

**Comment by BoogieMonster1O1** [comment]
> Bad TMh5

🤔

**Comment by BoogieMonster1O1** [comment]
> Stop necroing. Lock thread pls

psyche
only contributors are necroing

**Comment by BoogieMonster1O1** [comment]
> Bad tm

h5

**Pull request merged**